src/stirling/source_connectors/tcp_stats/tcp_stats_bpf_test.cc

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2018- The Pixie Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <string>
+
+#include "src/common/base/base.h"
+#include "src/common/exec/exec.h"
+#include "src/common/testing/testing.h"
+#include "src/shared/types/types.h"
+#include "src/stirling/core/data_table.h"
+#include "src/stirling/source_connectors/tcp_stats/tcp_stats_connector.h"
+#include "src/stirling/source_connectors/tcp_stats/tcp_stats_table.h"
+#include "src/stirling/source_connectors/tcp_stats/testing/tcp_stats_bpf_test_fixture.h"
+#include "src/stirling/testing/common.h"
+
+namespace px {
+namespace stirling {
+
+using ::px::stirling::testing::RecordBatchSizeIs;
+using ::px::stirling::testing::TcpTraceBPFTestFixture;
+using ::testing::ContainsRegex;
+using ::testing::HasSubstr;
+
+class TcpTraceTest : public TcpTraceBPFTestFixture {};
+
+//-----------------------------------------------------------------------------
+// Test Scenarios
+//-----------------------------------------------------------------------------
+
+TEST_F(TcpTraceTest, Capture) {
+  StartTransferDataThread();
+
+  // Send "hello" to 127.0.0.1, total 6 bytes of data
+  std::string cmd1 = "echo \"hello\" | nc 127.0.0.1 22";
```

Comment:
We are in the process of moving our tests to a qemu setup (#709). Since that setup will boot custom kernels, I validated that this test passes in our existing setup and that qemu test runner.

```
ddelnano@vigenere:~/code/pixie ((421e489ede...)) $ bazel test -c opt --config=qemu-bpf --cache_test_results=no --remote_download_outputs=all //src/stirling/source_connectors/tcp_stats:tcp_stats_bpf_test --test_output=streamed   --//bazel/test_runners/qemu_with_kernel:kernel_version=latest
WARNING: option '--build_tag_filters' was expanded to from both option '--config=qemu-bpf' (source command line options) and option '--config=qemu-bpf' (source command line options)
WARNING: option '--test_tag_filters' was expanded to from both option '--config=qemu-bpf' (source command line options) and option '--config=qemu-bpf' (source command line options)
INFO: Invocation ID: 8703db7a-078b-48d8-ae91-df8e01b1bd6b
INFO: Streaming build results to: https://bb.corp.pixielabs.ai/invocation/8703db7a-078b-48d8-ae91-df8e01b1bd6b
WARNING: option '--build_tag_filters' was expanded to from both option '--config=qemu-bpf' (source command line options) and option '--config=qemu-bpf' (source command line options)
WARNING: option '--test_tag_filters' was expanded to from both option '--config=qemu-bpf' (source command line options) and option '--config=qemu-bpf' (source command line options)
WARNING: Streamed test output requested. All tests will be run locally, without sharding, one at a time
INFO: Analyzed target //src/stirling/source_connectors/tcp_stats:tcp_stats_bpf_test (44 packages loaded, 324 targets configured).
INFO: Found 1 test target...
Formatting '/tmp/tmp.NrPjk9wIXX.qcow2', fmt=qcow2 cluster_size=65536 extended_l2=off compression_type=zlib size=4294967296 backing_file=/home/ddelnano/.cache/bazel/_bazel_ddelnano/b2e8b093089e374aa7d44dfed4f9953b/sandbox/linux-sandbox/418/execroot/px/bazel-out/k8-opt/bin/src/stirling/source_connectors/tcp_stats/tcp_stats_bpf_test.runfiles/px/bazel/test_runners/qemu_with_kernel/disk_image.qcow2 backing_fmt=qcow2 lazy_refcounts=off refcount_bits=16
Could not access KVM kernel module: No such file or directory
qemu-system-x86_64: failed to initialize kvm: No such file or directory
Target //src/stirling/source_connectors/tcp_stats:tcp_stats_bpf_test up-to-date:
bazel-bin/src/stirling/source_connectors/tcp_stats/tcp_stats_bpf_test
INFO: Elapsed time: 122.832s, Critical Path: 116.53s
INFO: 32 processes: 1 internal, 31 linux-sandbox.
//src/stirling/source_connectors/tcp_stats:tcp_stats_bpf_test            PASSED in 0.8s
 
Executed 1 out of 1 test: 1 test passes.
There were tests whose specified size is too big. Use the --test_verbose_timeout_warnings command line option to see which ones these are.
INFO: Streaming build results to: https://bb.corp.pixielabs.ai/invocation/8703db7a-078b-48d8-ae91-df8e01b1bd6b
```

Reply:
FYI this test didn't actually pass, because it was run before no KVM caused a test failure, see the logs. I think this test doesn't work under QEMU b/c `nc` is not in the sysroot.